ethstorage/p2p/protocol/syncclient.go

```diff
@@ -731,7 +732,10 @@ func (s *SyncClient) assignBlobRangeTasks() {
 
 				if err != nil {
 					if e, ok := err.(*yamux.Error); ok && e.Timeout() {
-						log.Debug("Failed to request blobs", "peer", pr.id.String(), "err", err)
+						log.Debug("Request blobs timeout", "peer", pr.id.String(), "err", err)
+					} else if returnCode == streamError && strings.Contains(err.Error(), "no addresses") {
+						pr.isBad = true
```

Comment:
Does Prysm have a similar issue to ours, and how do they handle it?

Reply:
Prysm does not add addresses to peerStore, but directly connect to peer using AddrInfo.
![a343d35f97d364847231b9798a7e4cb](https://github.com/ethstorage/es-node/assets/14110556/46ae30a4-18de-4fce-aca5-1f5cacf48af3)
![6710635e392f82458cdc41cb35fe7bd](https://github.com/ethstorage/es-node/assets/14110556/79a108b1-b248-45ee-bae8-90c60a5a5a94)